File: cpp/485_Max_Consecutive_Ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxCount = 0;
        int tempCount = 0;
        
        for (int num : nums) {
            if (num == 0) {
                maxCount = max(maxCount, tempCount);
                tempCount = 0;
            } else {
                tempCount++;
            }
        }
        maxCount = max(maxCount, tempCount);
        
        return maxCount;
    }
};

File: rust/485_Max_Consecutive_Ones.rs
use std::cmp;

impl Solution {
    pub fn find_max_consecutive_ones(nums: Vec<i32>) -> i32 {
        let mut maxCount: i32 = 0;
        let mut tempCount: i32 = 0;
        for num in nums {
            if (num == 0) {
                maxCount = cmp::max(maxCount, tempCount);
                tempCount = 0;
            } else {
                tempCount += 1;
            }
        }
        maxCount = cmp::max(maxCount, tempCount);

        return maxCount;
    }
}
